bayes_kit/rhat.py

```diff
@@ -40,3 +120,61 @@ def rhat(chains: list[SeqType]) -> FloatType:
         + np.var(means, ddof=1) / np.mean(vars)
     )
     return r_hat
+
+
+def split_rhat(chains: list[SeqType]) -> FloatType:
+    """
+    Return the potential scale reduction factor (R-hat) for a list of
+    Markov chains consisting of each of the input chains split in half.
+    Unlike the base `rhat(chains)` function, this version is applicable
+    to a single Markov chain.
+
+    Split R-hat was introduced in the *Stan Reference Manual.*  The
+    first official publication was in the following book.
+
+    Gelman, A., Carlin, J.B., Stern, H.S., Dunson, D.B., Vehtari,
+    A. and Rubin, D.B., 2013. *Bayesian Data Analysis.* Third Edition.
+    CRC press.
+
+    See :func:`split_chains` for a specification of splitting.
+
+    Parameters:
+    chains: List of univariate Markov chains.
+
+    Returns:
+    Split R-hat statistic.
+
+    Throws:
+    ValueError: If there are fewer than two chains or if any chain has
+    fewer than than four elements.
+    """
+    return rhat(split_chains(chains))
+
+
+def rank_normalized_rhat(chains: list[SeqType]) -> FloatType:
+    """Return the rank-normalized R-hat for the specified chains.
+    Rank normalized r-hat replaces each value in the chains with its
+    rank, applies a shifted inverse standard normal cdf, and
+    returns the split R-hat value of the result.
+
+    Rank-normalized R-hat was introduced in the following paper.
```

Comment:
Might be good to include a sentence or two about why you'd want to do this instead of normal r-hat. (I know there's a paper cite, but that's a pretty big detour.)

Reply:
I added a one-sentence explanation for each, but I don't know how to say anything meaningful without a lot of stats jargon (mean, variance, distribution, stationary distribution, Markov chain, normality, trend, Cauchy distribution).

>  Rank-normalized R-hat should be more robust in situations where the stationary distribution of the Markov chains is not normal (e.g., for density targets for which means and/or variances are undefined, such as the Cauchy distribution).

> The main utility of splitting is to diagnose non-stationary chains  (e.g., ones with an upward or downward trend).  Unlike the base `rhat(chains)` function, this version is applicable to a single Markov chain.

Reply:
That seems clear to me!